include/ginkgo/core/reorder/rcm.hpp

```diff
@@ -109,7 +109,7 @@ class Rcm
      *
      * @return the permutation (permutation matrix)
      */
-    std::shared_ptr<const PermutationMatrix> get_permutation() const
+    std::shared_ptr<const LinOp> get_permutation() const override
```

Comment:
This change is an actual interface break.
Previously, you could have called
```
rcm->get_permutation()->get_permutation_size();
```
However, now, you need to do it with a cast since you no longer know the actual type of the permutation:
```
as<matrix::Permutation>(rcm->get_permutation())->get_permutation_size();
```
Currently, this affects only RCM (as there is no other reordering).
Is it necessary to make the permutation type a LinOp (meaning are you expecting other permutation types)?